core/state.py

```diff
@@ -2288,6 +2288,15 @@ def Unset(self, lval, which_scopes):
                         raise error.Runtime(
                             "%s[%d]: Index is out of bounds for array of length %d"
                             % (var_name, lval.index, n))
+                elif val.tag() == value_e.SparseArray:
+                    val = cast(value.SparseArray, UP_val)
+                    error_code = bash_impl.SparseArray_UnsetElement(
+                        val, mops.IntWiden(lval.index))
+                    if error_code == 1:
```

Comment:
Yeah since we are using this error code a lot, I think it's good to call it `bash_impl.IndexOutOfRange`